numpy/ma/core.py

```diff
@@ -328,19 +343,7 @@ def minimum_fill_value(obj):
     inf
 
     """
-    errmsg = "Unsuitable type for calculating minimum."
-    if hasattr(obj, 'dtype'):
-        return _recursive_extremum_fill_value(obj.dtype, min_filler)
-    elif isinstance(obj, float):
-        return min_filler[ntypes.typeDict['float_']]
-    elif isinstance(obj, int):
-        return min_filler[ntypes.typeDict['int_']]
-    elif isinstance(obj, long):
-        return min_filler[ntypes.typeDict['uint']]
```

Comment:
The old behavior is arguably a bug since it may violate the docstring of the various sorting functions. 

Eg, the description of the `endswith` argument of `argsort`  says "Whether missing values (if any) should be treated as the largest values (True) or the smallest values (False)". For long types, the `mininum_fill_value` behavior was not giving the actual smallest values.

I for one am ok with the change as long as it's listed in release notes.